src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -379,16 +379,5 @@ private static bool TryDecodeSuppressMessageAttributeData(AttributeData attribut
 
             return true;
         }
-
-        internal enum TargetScope
```

Comment:
@dotnet/roslyn-compiler Only compiler change in the PR is moving this nested internal enum to a separate file, so it can be used by the added analyzer to detect invalid SuppressMessageAttribute target/scope. This new file is linked into the IDE analyzer layer.